File: C++/Heliocentric.cpp
#include <iostream>
#include <string>

int input(std::string msg){
	int x = 0;
	std::cout << msg;
	std::cin >> x;
	return x;
}

int main(){
	int earthDays = input("Enter input for Earth: ");
	int marsDays = input("Enter input for Mars: ");
	int counter = 0;
	while (true){
		if (marsDays % 687 == 0 && earthDays % 365 == 0) break;
		counter++; marsDays++; earthDays++;
	}
	std::cout << counter << std::endl;
}

File: Python/Heliocentric.py
earth_days = input("Enter input for Earth: ")
mars_days = input("Enter input for Mars: ")
counter = 0
while (True):
	if mars_days % 687 == 0 and earth_days % 365 == 0:
		break
	counter += 1; earth_days += 1; mars_days += 1;

print counter